pkg/daemon/ceph/osd/volume.go

```diff
@@ -974,6 +974,11 @@ func GetCephVolumeRawOSDs(context *clusterd.Context, clusterInfo *client.Cluster
 				}
 
 				target := oposd.EncryptionDMName(pvcName, oposd.DmcryptBlockType)
+				// remove stale dm device left by previous OSD.
+				err = removeEncryptedDevice(context, target)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to remove stale dm device %q", target)
```

Comment:
> What if we just log the error and continue? There seems like a chance in the future the error "No such device" could change (error checks on strings aren't always reliable), then this could be broken if we fail this. The worst that happens is that the open will fail anyway.

This sounds good to me, 
I've removed strings checks and warning logging the error now. 

@leseb Does this sound fine to you too?

Reply:
I don't think the return code `ENXIO` will ever change. Ideally, we would look deeper into the executor code to effectively return a known return code so that we can check it.

I don't intend to hold this PR though, so log warning from the caller is fine.